cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -102,9 +102,54 @@ func (p *BicepProvider) Initialize(ctx context.Context, projectPath string, opti
 // values are unset.
 //
 // An environment is considered to be in a provision-ready state if it contains both an AZURE_SUBSCRIPTION_ID and
-// AZURE_LOCATION value.
+// AZURE_LOCATION value. Additionally, for resource group scoped deployments, an AZURE_RESOURCE_GROUP value is required.
 func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
-	return EnsureSubscriptionAndLocation(ctx, p.env, p.prompters)
+	if err := EnsureSubscriptionAndLocation(ctx, p.env, p.prompters); err != nil {
+		return err
+	}
+
+	modulePath := p.modulePath()
+	if _, err := os.Stat(modulePath); errors.Is(err, os.ErrNotExist) {
+		// If there's not template, just behave as if we are in a subscription scope (and don't ask about
+		// AZURE_RESOURCE_GROUP). Future operations which try to use the infrastructure may fail, but that's ok. These
+		// failures will have reasonable error messages.
+		//
+		// We want to handle the case where the provider can `Initialize` even without a template, because we do this
+		// in a few of our end to end telemetry tests to speed things up.
+		return nil
+	}
+
+	_, template, err := p.compileBicep(ctx, modulePath)
+	if err != nil {
+		return fmt.Errorf("compiling bicep template: %w", err)
+	}
+
+	scope, err := template.TargetScope()
+	if err != nil {
+		return err
+	}
+
+	if scope == azure.DeploymentScopeResourceGroup {
+		if !p.alphaFeatureManager.IsEnabled(ResourceGroupDeploymentFeature) {
+			return ErrResourceGroupScopeNotSupported
+		}
+
+		p.console.WarnForFeature(ctx, ResourceGroupDeploymentFeature)
+
+		if p.env.Getenv(environment.ResourceGroupEnvVarName) == "" {
+			rgName, err := p.prompters.PromptResourceGroup(ctx)
```

Comment:
IIRC, PromptResourceGroup() won't work in `--no-prompt` mode, as if has no default selections.

Do we want to fail here if we are running in `--no-prompt`?   (or maybe within the PromptResourceGroup() fn?)

Reply:
> Do we want to fail here if we are running in --no-prompt? (or maybe within the PromptResourceGroup() fn?)

When `--no-prompt` is used, we register a special prompter either accepts the default response or fails if there is no default resonse.  So this will fail out of the box with an error like:

```
"error selecting resource group: no default response for prompt 'Pick a resource group to use: '"
```

https://github.com/Azure/azure-dev/blob/4a27d946981ca37ceb17e6df4a785a864048f8f9/cli/azd/pkg/input/asker.go#L19-L67